api/src/main/java/org/pmiops/workbench/elasticsearch/ElasticSearchService.java

```diff
@@ -0,0 +1,116 @@
+package org.pmiops.workbench.elasticsearch;
+
+import com.google.cloud.bigquery.FieldValue;
+import com.google.cloud.bigquery.QueryJobConfiguration;
+import com.google.cloud.bigquery.TableResult;
+import jnr.ffi.annotations.Synchronized;
+import org.apache.http.HttpHost;
+import org.elasticsearch.action.admin.cluster.health.ClusterHealthRequest;
+import org.elasticsearch.action.admin.cluster.health.ClusterHealthResponse;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.pmiops.workbench.api.BigQueryService;
+import org.pmiops.workbench.cohortbuilder.ParticipantCounter;
+import org.pmiops.workbench.cohortbuilder.ParticipantCriteria;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.inject.Provider;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+
+@Service
+public class ElasticSearchService {
+
+  private static final Logger log = Logger.getLogger(ElasticSearchService.class.getName());
+  private RestHighLevelClient client;
+  private Provider<WorkbenchConfig> configProvider;
+  private BigQueryService bigQueryService;
+  private ParticipantCounter participantCounter;
+
+  @Autowired
+  public ElasticSearchService(Provider<WorkbenchConfig> configProvider,
+                              BigQueryService bigQueryService,
+                              ParticipantCounter participantCounter) {
+    this.configProvider = configProvider;
+    this.bigQueryService = bigQueryService;
+    this.participantCounter = participantCounter;
+  }
+
+  /**
+   * This method will generate counts per the
```

Comment:
nit: this is some aggressive wrapping - are you optimizing for mobile IDE users?